lightning/src/ln/channelmanager.rs

```diff
@@ -2815,10 +2815,10 @@ enum NotifyOption {
 /// We allow callers to either always notify by constructing with `notify_on_drop` or choose to
 /// notify or not based on whether relevant changes have been made, providing a closure to
 /// `optionally_notify` which returns a `NotifyOption`.
-struct PersistenceNotifierGuard<'a, F: FnMut() -> NotifyOption> {
+struct PersistenceNotifierGuard<'a, F: FnOnce() -> NotifyOption> {
 	event_persist_notifier: &'a Notifier,
 	needs_persist_flag: &'a AtomicBool,
-	should_persist: F,
+	should_persist: Option<F>,
```

Comment:
The explanation why this needs to be an option could be helpful in the code rather than in the commit message.